torch/_inductor/scheduler.py

```diff
@@ -2030,6 +2030,25 @@ def can_fuse_vertical(self, node1, node2):
             if node1_names & self.name_to_fused_node[name].ancestors:
                 why("intermediate nodes between node1 & node2")
                 return False
+
+        # similar to can_inplace, if we are going to fuse a write subsequent to a read
+        # require that the indexing and size is the same
+        for write in node2.read_writes.writes:
```

Comment:
Does it matter for perf that this is n^2? I have no context on how many iterations we're talking about here.

Reply:
fair question, this is using same pattern already in this function but I can refactor both to make them O(n) (no compilation increase showed up in benchmark)

Reply:
I defer to you and whether it's worth it to refactor